Add AGS10 Sensor

mak-42 opened the pull request:
## Description:

Documentation for AGS10 TVOC Sensor.

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6070

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

mak-42 committed:
Add AGS10 Sensor

mak-42 committed:
Add AGS10 Sensor to the index.

mak-42 committed:
Fix issues with documentation of AGS10 Sensor.

mak-42 committed:
Fix issues with documentation of AGS10 Sensor.

mak-42 committed:
Fix issues with documentation of AGS10 Sensor.

mak-42 committed:
Fix issues with documentation of AGS10 Sensor.

mak-42 committed:
Fix issues with documentation of AGS10 Sensor.

mak-42 committed:
Fix issues with documentation of AGS10 Sensor.

jesserockz commented on components/sensor/ags10.rst:
```diff
@@ -0,0 +1,130 @@
+
+  - **name** (**Required**, string): The name for the tvoc sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
```
We don't duplicate config lines that are linked in sensor core docs
```suggestion
- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
  All options from :ref:`Sensor <config-sensor>`.
```

mak-42 replied:
Done.

jesserockz commented on components/sensor/ags10.rst:
```diff
@@ -0,0 +1,130 @@
+        address: 0x1A
+        update_interval: 60s
+
+Configuration variables:
```
`address` and `resistance` sensors are missing from the documentation.

mak-42 replied:
`version` and `resistance`. The `address` was already described.
Done.

jesserockz commented on components/sensor/ags10.rst:
```diff
@@ -0,0 +1,130 @@
+
+.. code-block:: yaml
+
+    # I²C Bus configuration
+    i2c:
+      sda: GPIO21
+      scl: GPIO22
+      scan: true
+      frequency: 15kHz
+
+    # Example configuration entry
+    sensor:
+      - platform: ags10
+        tvoc:
+          name: TVOC
+        address: 0x1A
+        update_interval: 60s
```
Example YAML should be basic to allow the user to read the documentation and understand how to configure it
```suggestion
    # Example configuration entry
    sensor:
      - platform: ags10
        tvoc:
          name: TVOC
```

mak-42 replied:
Done

mak-42 committed:
Merge branch 'next' into ags10

mak-42 committed:
Fix the issues pointed by reviewer:
1. Duplicated config lines were removed.
2. The description of "version" and "resistance" sensors are added.

mak-42 committed:
Fix the issues pointed by reviewer:
1.The configuration example was simplified.

mak-42 committed:
Merge branch 'next' into ags10

mak-42 committed:
Merge branch 'next' into ags10

mak-42 committed:
Merge branch 'next' into ags10

mak-42 committed:
Merge branch 'next' into ags10

jesserockz commented on components/sensor/ags10.rst:
```diff
@@ -0,0 +1,119 @@
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
+  sensor. Defaults to ``60s``.
+- **version** (*Optional*, int): The firmware version of the sensor.
+- **resistance** (*Optional*, int): The initial value of the sesor resistance.
```
```suggestion
- **resistance** (*Optional*, int): The initial value of the sensor resistance.
```

jesserockz committed:
Update components/sensor/ags10.rst

jesserockz commented on components/sensor/ags10.rst:
```diff
@@ -0,0 +1,119 @@
+  the sensor. Defaults to ``0x1A``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
+  sensor. Defaults to ``60s``.
+- **version** (*Optional*, int): The firmware version of the sensor.
+- **resistance** (*Optional*, int): The initial value of the sensor resistance.
```
```suggestion
- **version** (*Optional*): The firmware version of the sensor.
  All options from :ref:`Sensor <config-sensor>`.
- **resistance** (*Optional*): The initial value of the sensor resistance.
  All options from :ref:`Sensor <config-sensor>`.
```

jesserockz commented on components/sensor/ags10.rst:
```diff
@@ -0,0 +1,119 @@
+------------------------
+
+- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
+  All other options from :ref:`Sensor <config-sensor>`.
```
```suggestion
  All options from :ref:`Sensor <config-sensor>`.
```

jesserockz committed:
Apply suggestions from code review

jesserockz commented on components/sensor/ags10.rst:
```diff
@@ -0,0 +1,121 @@
+ESPHome. The :ref:`I²C Bus <i2c>` is
+required to be set up in your configuration for this sensor to work.
+
+.. warning::
```
A warning is a bit scary for this
```suggestion
.. note::
```

jesserockz committed:
Update components/sensor/ags10.rst
